tools/bridge/bridge/node_status.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from typing import Dict, List, Optional
+
+import attr
+import gevent
+import tenacity
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class NodeStatus:
+    is_syncing: bool
+    block_number: int
+    latest_synced_block: int
+    syncing_map: Dict
+    client_version: str
+    is_light_node: Optional[bool]
+    block_gap: List[int]
+
+
+def get_node_status_parity(w3):
+    client_version = w3.clientVersion
+    block_number = w3.eth.blockNumber
+
+    syncing_map = w3.eth.syncing or None
+
+    chain_status = w3.manager.request_blocking("parity_chainStatus", [])
+    node_kind = w3.manager.request_blocking("parity_nodeKind", [])
+
+    if chain_status.blockGap is not None:
+        block_gap = [int(x, 16) for x in chain_status.blockGap]
+    else:
+        block_gap = None
+
+    is_light_node = node_kind.capability == "light"
+
+    if block_gap and not is_light_node:  # warp mode syncing
```

Comment:
Why do you not just have `if block_gap`?

Reply:
the light mode node always has a block gap over the full range of blocks.

Reply:
ah ok